### PR TITLE
docs: refine function usages

### LINKED ISSUE #664: Use either of `eventHandler` and `defineEventHandler` consistently in documentation
### Describe the change

It is said in [event-handler page](https://h3.unjs.io/guide/event-handler) that, 

>  You can use `eventHandler` and `defineEventHandler` interchangeably. They are aliases. You can use the one you prefer but stick to it for consistency.

In this point, h3 docs doesn't stick to either of them.
Although documentation and its sample code are not actual application code, 
it might be better to follow the convention that it tells by itself.

If @pi0 and/or @Barbapapazes tell me which to go, I will submit a PR to unify usages, 
or please close this issue if it's fine to use both for some reasons.

### URLs

All pages where `eventHandler` and `defineEventHandler` are used.

### Additional information

- [X] Would you be willing to help?

### PR DESCRIPTION
**Pull request by NozomuIkuta**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Resolves #664

As @Barbapapazes suggested, I replace `eventHandler` calls with `defineEventHandler` because the latter is more used.
At the same time, I also fixed the following points:

- Sort imports in alphabetical order
  - In this point, it is better to fix #663 for consistency
- Fix typos
  - Typo fixed by #658 is included in this PR too (so, it should be merged first)
- Refine sample codes

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/1.guide/2.event-handler.md [hunk 1]
```diff
@@ -66,38 +64,36 @@ Any of above values could also be wrapped in a [`Promise`](https://developer.moz
...
-    });
-  }),
-);
+app.use(defineEventHandlers(async (event) => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve({ url: event.node.req.url });
+    }, 1000);
+  });
+}));
```
Comment (NozomuIkuta) [inline docs/1.guide/2.event-handler.md]: As far as I saw all pages, it seems that line break between `app.use(` and `defineEventHandler` is inserted only when first argument is path string:

```ts
app.use(
  "/",
  defineEventHandlers(async (event) => {
    // ...
  }),
)
```

So, for consistency, I remove line break in some sample code:

```ts
app.use(defineEventHandlers(async (event) => {
  // ...
}))
```

File: docs/1.guide/4.event.md [hunk 2]
```diff
@@ -13,28 +13,26 @@ An event is passed through all the lifecycle hooks and composable utils to use i
...
-    // Echo back request as response
-    return {
-      path: event.path,
-      method: event.method,
-      query,
-      body,
-    }
-  }),
-);
+import { defineEventHandler, getQuery, readBody } from "h3";
```
Comment (NozomuIkuta) [inline docs/1.guide/4.event.md:16]: This typo (`query` 👉 `getQuery`) had been already fixed in #658, before I recognized it.

File: docs/1.guide/4.event.md [hunk 3]
```diff
@@ -108,15 +106,11 @@ You must craft a response using the [`Response`](https://developer.mozilla.org/e
...
-  defineEventHandler((event) => {
-    await event.respondWith(new Response("Hello World"));
+defineEventHandler((event) => {
+  await event.respondWith(new Response("Hello World"));
 
-    return "..."; // DOES NOT WORKS
-  }),
-);
+  return "..."; // DOES NOT WORKS
+});
```
Comment (NozomuIkuta) [inline docs/1.guide/4.event.md:113]: This might be conflicted with #662

**Review by pi0 — APPROVED** [review]
Thanks ❤️

**Pull request merged**